rust/worker/src/server.rs

```diff
@@ -735,197 +473,48 @@ mod tests {
     }
 
     #[tokio::test]
-    #[cfg(debug_assertions)]
-    async fn validate_get_vectors_request() {
-        use chroma_proto::vector_reader_client::VectorReaderClient as Client;
-        use chroma_types::chroma_proto::GetVectorsRequest as Request;
-
-        let mut reader = Client::connect(run_server()).await.unwrap();
-
-        let first_request = Request {
-            ids: vec![],
-            segment_id: SEGMENT_UUID.to_string(),
-            collection_id: COLLECTION_UUID.to_string(),
-            version_context: Some(RequestVersionContext {
-                collection_version: 0,
-                log_position: 0,
-            }),
+    async fn validate_count_plan() {
+        let mut reader = QueryExecutorClient::connect(run_server()).await.unwrap();
+        let mut scan_operator = scan();
+        let request = chroma_proto::CountPlan {
+            scan: Some(scan_operator.clone()),
         };
+
         // segment or collection not found
-        let request = first_request.clone();
-        let response = reader.get_vectors(request).await;
+        let response = reader.count(request).await;
         assert_eq!(response.unwrap_err().code(), tonic::Code::NotFound);
 
-        // invalid collection uuid
```

Comment:
this case got dropped

Reply:
just consider testing a bit more

Reply:
Added a few more cases in the test including this one